plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -57,6 +58,7 @@ export interface RouterOptions {
   taskWorkers?: number;
   containerRunner?: ContainerRunner;
   taskBroker?: TaskBroker;
+  nunjucksFilters?: Record<string, NunjucksFilter>;
```

Comment:
Maybe just `additionalTemplateFilters` :) for some reason I really really dislike having nunjucks as a word in there. That would imply "this argument only works for some runners and is ignored by all others". `SecureTemplater.loadRenderer` calls the argument `additionalFilters` too, which aligns well.

Part of the problem here is the mega-service-building pattern. It just is crying out for a backend system redesign :) Until then, `additionalFilters` or `additionalTemplateFilters` is probably fine

Reply:
If we do not include "nunjucks" in the name, where is the appropriate place to document that this argument will not apply to the template pre-v1beta3?

Reply:
I'll let the other maintainers (@benjdlambert may have opinions but is out right now) chime in but in general i think that

- We're moving away from pre-beta3 pretty rapidly so the concern may be fairly small in practice
- It's actually not so much about just beta2 or beta3, it's that it only fits with certain _actions_ even, within templates, so instead it might better be named `fetchTemplateActionAdditionalFilters` or something like that ... nunjucks is an implementation detail isn't it

To me personally it'd be fine to leave with my proposed wording, or even `fetchTemplateActionAdditionalFilters` if we're unsure if it'll be translatable across several actions - and then in the future work towards changing entirely how this is wired together. Maybe one day it'll not be a mega-function that has an assortment of arguments for an even larger assortment of various concerns :) Instead maybe we'll provide at a granular level instead, passing in a `new FetchActionFactory({ additionalFilters: ... })`, clarifying the whole thing immensely

Reply:
As long as `app-template.yaml` uses these filters too (which I really need it to do), I don't think we can reasonably tie this variable name to the `fetch:template` action. Are there plans to move away from using a templating engine for `app-template.yaml`?

Reply:
It's also adds filters to the templating of the task spec itself, so more than just the fetch action. I'm gonna +1 `additionalTemplateFilters`

Reply:
I'd be fine with `additionalTemplateFilters`. _Slightly_ wordy, but it's not like anyone is going to have to type this out 100 times.

It gets my +1 (if `nunjucksTemplates` is off the table 😛 ).

Reply:
:shipit:

Reply:
Change is applied and push. `additionalTemplateFilters` is what I went with.

I'm going to mark this as resolved, but @Rugvip or others, please feel free to re-open the thread if you think more opinions are warranted.